### PR TITLE
MNT: update CI, numpy2

### PR DESCRIPTION
**Pull request by kmuehlbauer**

### CODE REVIEW

File: wradlib/georef/vector.py [hunk 1]
```diff
@@ -479,6 +479,6 @@ def get_centroid(polyg):
         x and y coordinate of the centroid
 
     """
-    if not type(polyg) == ogr.Geometry:
+    if type(polyg) is not ogr.Geometry:
```
Comment (mats-knmi) [inline wradlib/georef/vector.py:482]: These `is not` checks should be replaced with isinstance checks as well I think. Or do we want to be absolutely sure that `trg` is of type `Geometry` and not a subtype of `Geometry`?
```suggestion
    if not isinstance(polyg, ogr.Geometry):
```

File: wradlib/zonalstats.py [hunk 2]
```diff
@@ -340,7 +342,7 @@ def _get_intersection(self, *, trg=None, idx=None, buf=0.0):
             raise TypeError("Either `trg` or `idx` kwargs must be given!")
 
         # check for geometry
-        if not type(trg) == ogr.Geometry:
+        if type(trg) is not ogr.Geometry:
```
Comment (mats-knmi) [inline wradlib/zonalstats.py:345]: Same here
```suggestion
        if not isinstance(trg, ogr.Geometry):
```

File: wradlib/georef/vector.py (hunk 1 above)
Comment (kmuehlbauer) [reply wradlib/georef/vector.py:482]: Obviously missed those. Let's try it.

**Comment by kmuehlbauer** [comment]
@mats-knmi Thanks for engaging here. I'm going to merge this right after getting green lights. 

The other deprecation warnings can be handled in subsequent PR's.

The notebook-test is non-trivial to solve as `wetterdienst`-package latest version has diverged quite a bit from the version I've pinned (which only works on numpy 1, sigh). They are also pinning dependencies quite strict which prevents current usage. So need to think about that a bit.

**Comment by kmuehlbauer** [comment]
Most likely 0.49 wasn't pinned for numpy?

I do not like the strict pinning scheme over those many dependencies. In connection with conda- forge you end up in unresolvable environments pretty fast.

**Comment by mats-knmi** [comment]
Yeah strict dependencies just doesn't work with python since only one version of a package may be installed at once.

As far as I can see wetterdienst `0.49.0` also has a dependency on numpy < 2, so I don't understand how the github actions conda environment creation step can install these 2 dependencies side by side.

**Comment by kmuehlbauer** [comment]
I have no idea, the magic of GitHub. : grimacing:

**Pull request merged**